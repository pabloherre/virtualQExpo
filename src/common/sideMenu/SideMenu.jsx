import React from 'react';
import { connect } from 'react-redux';
import RoundedButton from '../buttons/RoundedButton/RoundedButton';
import { View } from 'react-native';
import AuthService from '../../modules/auth/Auth.service';
import UserService from '../../services/user/User.service';
import { colors } from '../../../theme';
import { FontAwesome, SimpleLineIcons } from '@expo/vector-icons';
import { SideMenuService } from './SideMenu.service';
import Typography from '../typography/Typography';
import { setTestInfo } from '../../utils/test.utils';

export class Menu extends React.Component {
  onPress = async () => {
    SideMenuService.closeMenu();
    await AuthService.logout();
    await UserService.setUser(null);
    this.props.navigation.navigate('Login');
  };

  render() {
    return (
      <View
        style={{
          flex: 1,
          alignItems: 'stretch',
          backgroundColor: colors.card
        }}
      >
        <View
          style={{
            alignItems: 'center',
            borderBottomWidth: 1,
            paddingVertical: 20,
            marginHorizontal: 20,
            borderBottomColor: colors.border
          }}
        >
          <FontAwesome name="user-circle" size={120} color={colors.secondary} style={{ paddingBottom: 20 }} />
          <Typography color="secondary" size="small">
            {this.props.user && this.props.user.email}
          </Typography>
        </View>
        <View style={{ flex: 1, marginVertical: 10, width: 200, alignSelf: 'center', justifyContent: 'flex-end' }}>
          <RoundedButton
            startIcon={<SimpleLineIcons name="logout" size={24} color="white" />}
            label="Logout"
            onPress={this.onPress}
            {...setTestInfo('testTouchableLogout')}
          />
        </View>
      </View>
    );
  }
}

const mapStateToProps = state => {
  return {
    user: state.user.data
  };
};

export default connect(mapStateToProps, null)(Menu);
