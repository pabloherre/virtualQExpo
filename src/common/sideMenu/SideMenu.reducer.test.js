import sideMenuReducer from './SideMenu.reducer';
import { CLOSE_MENU, OPEN_MENU } from './SideMenu.actions';

describe('SideMenu Reducer', () => {
  it('should should return initial state', async () => {
    const INITIAL_STATE = {
      open: false
    };

    expect(sideMenuReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle OPEN MENU action', async () => {
    const INITIAL_STATE = {
      open: true
    };

    expect(sideMenuReducer(INITIAL_STATE, { type: OPEN_MENU })).toMatchSnapshot();
  });

  it('should handle CLOSE MENU action', async () => {
    const INITIAL_STATE = {
      open: false
    };

    expect(sideMenuReducer(INITIAL_STATE, { type: CLOSE_MENU })).toMatchSnapshot();
  });
});
