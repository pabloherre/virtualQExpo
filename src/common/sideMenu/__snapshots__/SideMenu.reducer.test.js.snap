// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu Reducer should handle CLOSE MENU action 1`] = `
Object {
  "open": false,
}
`;

exports[`SideMenu Reducer should handle OPEN MENU action 1`] = `
Object {
  "open": true,
}
`;

exports[`SideMenu Reducer should should return initial state 1`] = `
Object {
  "open": false,
}
`;
